### ISSUE TITLE
why the code is not consistent with the paper

### ISSUE THREAD

**Issue by poisonwine** [ryanxhr/IVR#1]
the code update the critic in SQL is  'sp_term = (q-v) / alpha + 0.5', 
but according to  the paper, it should be 'sp_term = (q-v) / (2*alpha )+ 1 '.
And the policy improvement is also not consistent with the paper, see actor.py from  line 11 to line 17

‘’‘
v = value(batch.observations)
    q1, q2 = critic(batch.observations, batch.actions)
    q = jnp.minimum(q1, q2)

    weight = q - v
    weight = jnp.maximum(weight, 0)
    weight = jnp.clip(weight, 0, 100.)
’‘’

**Comment by xryoung** [comment]
> 

the same question here

**Comment by xryoung** [comment]
> the code update the critic in SQL is 'sp_term = (q-v) / alpha + 0.5', but according to the paper, it should be 'sp_term = (q-v) / (2*alpha )+ 1 '. And the policy improvement is also not consistent with the paper, see actor.py from line 11 to line 17
> 
> ‘’‘ v = value(batch.observations) q1, q2 = critic(batch.observations, batch.actions) q = jnp.minimum(q1, q2)
> 
> ```
> weight = q - v
> weight = jnp.maximum(weight, 0)
> weight = jnp.clip(weight, 0, 100.)
> ```
> 
> ’‘’

the same question here

**Comment by ryanxhr** [comment]
Thanks for pointing out that! 

### About the code to update the critic in SQL: 
Yes, it should be 'sp_term = (q-v) / alpha + 0.5'. We have updated our code to revise the bug. 

### About the code to update the actor: 
SQL and EQL extract the policy by Eq.(14) and Eq.(17), respectively. However, in practice, we found the scale of $\alpha$ does affect the performance as the behavior cloning weight involves $(Q-V) / \alpha$. To make the residual less affected by the scale of $\alpha$, we multiply a constant value to the residual, this can be seen as a kind of reward normalization and doesn't change the optimal policy. In EQL, we multiply 10 to the residual term and extract the policy by $\max_\pi \ E_{(s, a) \sim \mathcal{D}} \left[ \exp \left(10 \cdot \left(Q(s, a) - V(s) \right) / \alpha \right) \log \pi(a|s) \right]$. In SQL, we remove the "1+" term in Eq.(14) (this amounts to multiple a large value to the residual term) and extract the policy by $\max_{\pi} \ E_{(s, a) \sim \mathcal{D}}\left[1 \left(Q(s, a) - V(s) >0 \right) \left(Q(s, a) - V(s) \right) \log \pi(a|s)\right]$. Note that XQL (exactly the same as EQL) also applies a similar trick, it extracts the policy by using temperature $\beta$ (3.0 for MuJoCo tasks and 10.0 for AntMaze tasks) instead of $\alpha$.

We have updated the results and added the above clarification to our newest paper, it should be available on Arxiv in a few days. Please check out the paper and code for details.

**Comment by ryanxhr** [comment]
I will close this issue but feel free to re-open it if you have further questions!
